feature-utils/poly-look/test/react-components/legends.test.js

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import "@testing-library/jest-dom";
+import {
+  BlockLegend,
+  LineLegend,
+} from "../../src/react-components/legends.jsx";
+
+/**
+ * @jest-environment jsdom
+ */
+const legendValues = [
+  ["No color", "#000000"],
+  ["Gray", "#888888"],
+  ["All colors", "#ffffff"],
+];
+const legendEntries = legendValues.map((l) => {
+  return { description: l[0], color: l[1] };
+});
+const blockLegend = <BlockLegend legend={legendEntries} />;
+const lineLegend = <LineLegend legend={legendEntries} />;
+
+describe("Basic tests BlockLegend", () => {
+  it("Creates a BlockLegend component", () => {
+    const renderedBlockLegend = render(blockLegend);
+    expect(renderedBlockLegend.container).toBeTruthy();
+    expect(renderedBlockLegend.getByText("No color")).toBeInTheDocument();
+  });
+});
+
+describe("Basic tests LineLegend", () => {
+  it("Creates a LineLegend component", () => {
+    const renderedLineLegend = render(lineLegend);
+    expect(renderedLineLegend.container).toBeTruthy();
+    expect(renderedLineLegend.getByText("No color")).toBeInTheDocument();
+  });
```

Comment:
here on both tests, we `expect` same results 🤔 is there sth that distinguishes a `blockLegend` from a `lineLegend` ? 
(I might miss background here...but just by reading this seems a bit too vague)

Reply:
Yes, but I would need some more (or a lot) of instrumentation to distinguish between the two, mainly including DOM readers here. Might have to, anyway, but this was a minimal viable product, and it does give full coverage of the tested code (except for a couple of lines)

Reply:
Well, so it appears that semantically, `LineLegend` and `BlockLegend` are identical, correct? They only _look_ differently, which is probably not really wise to try and test manually. That makes me think if we couldn't do it like this:

1. Write a test suite we run against _both_ `BlockLegend` and `LineLegend`, without actually duplicating the test code (as it is right now).
2. Cover a little more functionality - try different legends inputs and e.g. ensure the number of entries is the expected amount, that kind of stuff.

Something like this:

```
describe("Legend component functionality tests", () => {
    const components = [ BlockLegend, LineLegend ];
    for (let component of components) {
        it(`Render ${component.name}`, () => {
            const rendered = render(<component legend={legendEntries} />);
            ....
        });
    }
});
```